java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java

```diff
@@ -3131,6 +3131,43 @@ public List<SystemOverview> getId(User loggedInUser, String name) {
         return SystemManager.listSystemsByName(loggedInUser, name);
     }
 
+    /**
+     * Get system IDs and last check in information for the given list of system names.
+     * @param loggedInUser The current user
+     * @param names of the servers
+     * @return Map Array containing maps with system Ids with the given name and the associated
+     * overview
+     *
+     * @apidoc.doc Get system IDs and last check in information for a given
+     * list of system names.
+     * @apidoc.param #session_key()
+     * @apidoc.param #array_single("string", "names")
+     * @apidoc.returntype
+     *      #return_array_begin()
+     *          #struct_begin("servers_list")
+     *              #prop_desc("string", "name", "The name of the queried system")
+     *              #prop_array_begin_desc("overview", "A list of systems found with the name")
+     *                  $SystemOverviewSerializer
+     *              #prop_array_end()
+     *          #struct_end()
+     *      #array_end()
+     */
+    @ReadOnly
+    public List<Map<String, Object>> getId(User loggedInUser, List<String> names) {
+
+        List<Map<String, Object>> result = new ArrayList<>();
+
+        for (String name : names) {
+            Map<String, Object> systemInfoMap = new HashMap<>();
+            List<SystemOverview> systemOverview = SystemManager.listSystemsByName(loggedInUser, name);
```

Comment:
I think this is not a good idea.
We will have customers adding 5000 names and this will result in 5000 calls of this query.

I wonder if you really need this as we have  "listSystems" which return 
id, name, last_checkin, created and last_boot.
According to the description it has everything what you want.

If you want to filter this list by name, we can add this method, but please use internally this function and do the filtering here in the code instead of calling multiple times a different function.

Reply:
@mcalmer when we talk about `name` we talk about a hostname or an FQDN?

Reply:
This name is the "profile name". It is the name you see also in the system list.